syntax = "proto3";

service OpenPoseEstimator {
    rpc estimate(Image) returns (DetectedPoses);
}

/*
 * Image to analyse
 * param data: bytes of the image file
 *             encoded in some image format
 */
message Image {
    bytes data = 1;
}

/*
 * Message to represent the discovered poses
 * param poses: list of detected poses
 */
message DetectedPoses {
    repeated Pose poses = 1;
}

/*
 * Message to represent a human pose
 * param key_points: detected keypoints for the pose
 */
message Pose {
    repeated KeyPoint key_points = 1;
}

/*
 * Message to represent a body part within
 * the pose representation
 * param index: index of the keypoint in the person skeleton
 * param x: x coordinate of the keypoint
 * param y: y coordinate of the keypoint
 * param confidence: confidence score for the keypoint
 */
message KeyPoint {
    uint32 index = 1;
    float x = 2;
    float y = 3;
    float score = 4;
}